### PR TITLE
feat: implement fetch user

### PR DESCRIPTION
**Pull request by alexandrabakhar**
Also add context, tanstack query for fetch user

### CODE REVIEW

File: src/app/layouts/MainLayout.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
+import { useUserContext } from '@/entities'
 import { EmptyPage, MainPage } from '@/pages'
 import { Header } from '@/widgets'
 
 export const MainLayout = () => {
-  const isUser = true
+  const { user, errorStatus } = useUserContext()
+
+  const contentType = errorStatus === '404' ? 'user' : 'initial'
```
Comment (remadyt) [inline src/app/layouts/MainLayout.tsx]: if 500 erorStatus?

File: src/features/search-user/ui/SearchForm.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { useUserContext } from '@/entities'
+import { baseQuery } from '@/shared/api/baseQuery'
+import { SearchIcon } from '@/shared/lib'
+import { Input } from '@/shared/ui'
+import { useMutation } from '@tanstack/react-query'
+import { useEffect } from 'react'
+type SearchFormValues = {
```
Comment (remadyt) [inline src/features/search-user/ui/SearchForm.tsx:8]: prettier work?

File: src/entities/user/model/UserContext.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { createContext } from 'react'
+import { User } from './types'
+
+type UserContextType = {
+  errorStatus: null | string
+  setErrorStatus: React.Dispatch<React.SetStateAction<string | null>>
+  isLoading: boolean
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
```
Comment (remadyt) [inline src/entities/user/model/UserContext.ts]: why need ``` React.```? use ```Dispatch<SetStateAction<boolean>>```

File: src/app/providers/UserContextProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { User } from '@/entities'
+import { UserContext } from '@/entities/user/model/UserContext'
```
Comment (remadyt) [inline src/app/providers/UserContextProvider.tsx]: fix imports here and all such places , as sample look. https://feature-sliced.design/ru/docs/guides/examples/auth

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/features/search-user/ui/SearchForm.tsx (hunk 2 above)
Comment (alexandrabakhar) [reply src/features/search-user/ui/SearchForm.tsx:8]: yes, but settings don't include rules about empty string after imports

File: src/app/layouts/MainLayout.tsx (hunk 1 above)
Comment (alexandrabakhar) [reply src/app/layouts/MainLayout.tsx]: So I need another page like 'everything was broken, wait a minute to continue'

Comment (alexandrabakhar) [reply src/app/layouts/MainLayout.tsx]: added defaultError content for empty block (for other errors)

File: src/entities/user/hooks/useUserContext.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { useCallback, useContext } from 'react'
+import { UserContext } from '@/entities'
+import { User } from '@/entities'
```
Comment (remadyt) [inline src/entities/user/hooks/useUserContext.tsx]: not need 2 imports from entities, add eslint rule for handle this

File: src/entities/user/model/UserContext.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Dispatch, SetStateAction, createContext } from 'react'
+import { User } from './types'
+
+type UserContextType = {
```
Comment (remadyt) [inline src/entities/user/model/UserContext.ts]: type not needed in naming

File: src/entities/user/model/UserContext.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  errorStatus: null | string
+  setErrorStatus: Dispatch<SetStateAction<string | null>>
+  user: User | null
```
Comment (remadyt) [inline src/entities/user/model/UserContext.ts]: implement Nullable helper, type Nullable<T> = T | null, and declare in global.d.ts

File: src/entities/user/ui/card/Card.tsx [hunk 8]
```diff
@@ -1,32 +1,37 @@
 import { Avatar, Link, Typography } from '@/shared/ui'
 import styles from './styles.module.css'
-import FollowersIcon from '@/entities/user-card/lib/followers.svg?react'
-import FollowingIcon from '@/entities/user-card/lib/following.svg?react'
-import mockAvatarSrc from '@/entities/user-card/lib/mockAvatar.png'
+import { FollowersIcon, FollowingIcon, useUserContext } from '@/entities/user'
```
Comment (remadyt) [inline src/entities/user/ui/card/Card.tsx]: from  @/entities

File: src/entities/user/ui/card/Card.tsx [hunk 9]
```diff
@@ -1,32 +1,37 @@
...
       </div>
 
       <div className={styles.infoContainer}>
         <div className={styles.infoBox}>
           <FollowersIcon />
           <Typography tagType='p' color='secondary'>
-            65.8k followers
+            {user?.followers} followers
           </Typography>
+          {/* TODO: сделать форматирование тысячи */}
```
Comment (remadyt) [inline src/entities/user/ui/card/Card.tsx]: why not implemented?

File: src/features/search-user/ui/SearchForm.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
+import { useUserContext } from '@/entities'
+import { baseQuery } from '@/shared/api/baseQuery'
```
Comment (remadyt) [inline src/features/search-user/ui/SearchForm.tsx]: from shared/api

File: src/features/search-user/ui/SearchForm.tsx [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        following,
+        name,
+        account_url: html_url,
+        repos_count: public_repos,
+      }
+      onSuccess(userData)
+    },
+  })
+
+  const handleOnSubmit: React.FormEventHandler<
```
Comment (remadyt) [inline src/features/search-user/ui/SearchForm.tsx]: remove ```React. ```

File: src/pages/EmptyPage/ui/EmptyPage.tsx [hunk 12]
```diff
@@ -3,9 +3,15 @@ import { EmptyBlock } from '@/widgets'
...
 
 type EmptyPageProps = {
-  contentType: 'initial' | 'user'
+  error?: string | null
 }
 
-export const EmptyPage: FC<EmptyPageProps> = ({ contentType }) => {
+export const EmptyPage: FC<EmptyPageProps> = ({ error }) => {
+  const contentType = error
+    ? error === '404'
```
Comment (remadyt) [inline src/pages/EmptyPage/ui/EmptyPage.tsx]: refactor, nested ternary - bad practice. use switch case or mapper pattern

File: src/pages/EmptyPage/ui/EmptyPage.tsx [hunk 13]
```diff
@@ -3,9 +3,15 @@ import { EmptyBlock } from '@/widgets'
 import { FC } from 'react'
 
 type EmptyPageProps = {
-  contentType: 'initial' | 'user'
+  error?: string | null
 }
 
-export const EmptyPage: FC<EmptyPageProps> = ({ contentType }) => {
+export const EmptyPage: FC<EmptyPageProps> = ({ error }) => {
```
Comment (remadyt) [inline src/pages/EmptyPage/ui/EmptyPage.tsx:9]: need all components type on 1 style. ```FC<Props> or ({}: Props)```

File: src/pages/MainPage/ui/MainPage.tsx [hunk 14]
```diff
@@ -1,12 +1,19 @@
...
+import { EmptyBlock } from '@/widgets'
 
 export const MainPage = () => {
+  const { user } = useUserContext()
+
   return (
     <main className={styles.main}>
       <Card />
-      <Repositories />
+      {user?.repos_count === 0 ? (
```
Comment (remadyt) [inline src/pages/MainPage/ui/MainPage.tsx]: move to const ````user?.repos_count === 0```

File: src/widgets/Repositories/api/fetchRepositories.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { RepositoryData } from '@/entities'
+import { baseQuery } from '@/shared/api/baseQuery'
+
+type FetchRepositories = ({
+  username,
+  perPage,
+  page,
+}: {
```
Comment (remadyt) [inline src/widgets/Repositories/api/fetchRepositories.ts]: move to separate type ```{
  username: string
  perPage: number
  page: number
}```

File: src/widgets/Repositories/hooks/useRepositories.tsx [hunk 16]
```diff
@@ -0,0 +1,30 @@
+import { useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { getRepoCountPerPage } from '@/widgets/Repositories/lib/getRepoCountPerPage'
+import { getTotalPages } from '@/widgets/Repositories/lib/getTotalPages'
```
Comment (remadyt) [inline src/widgets/Repositories/hooks/useRepositories.tsx]: fix imports

File: src/widgets/Repositories/hooks/useRepositories.tsx [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+import { User, useUserContext } from '@/entities'
+
+export const useRepositories = () => {
+  const repositoriesRef = useRef<HTMLUListElement | null>(null)
+  const [repoCountPerPage, setRepoCountPerPage] = useState(1)
```
Comment (remadyt) [inline src/widgets/Repositories/hooks/useRepositories.tsx]: do not use magic numbers

File: src/widgets/Repositories/ui/Repositories.tsx [hunk 18]
```diff
@@ -1,35 +1,57 @@
 import { Repository } from '@/entities'
 import styles from './styles.module.css'
 import { Typography } from '@/shared/ui'
-import { useEffect, useRef, useState } from 'react'
-import { getRepoCountPerPage } from '@/widgets/Repositories/lib/getRepoCountPerPage'
+import { useEffect, useState } from 'react'
+
```
Comment (remadyt) [inline src/widgets/Repositories/ui/Repositories.tsx]: remove empty line

File: src/widgets/Repositories/ui/Repositories.tsx [hunk 19]
```diff
@@ -1,35 +1,57 @@
...
 import { Pagination } from '@/features/pagination'
+import { useMutation } from '@tanstack/react-query'
+
+import { fetchRepositories } from '@/widgets/Repositories/api/fetchRepositories'
+import { useRepositories } from '@/widgets/Repositories/hooks/useRepositories'
 
 export const Repositories = () => {
   const [currentPage, setCurrentPage] = useState(1)
```
Comment (remadyt) [inline src/widgets/Repositories/ui/Repositories.tsx]: do not use magic numbers

File: src/widgets/Repositories/ui/Repositories.tsx [hunk 20]
```diff
@@ -1,35 +1,57 @@
...
 
   return (
     <div className={styles.container}>
       <Typography tagType='h1'>Repositories ({totalRepoCount})</Typography>
+
       <ul ref={repositoriesRef} className={styles.list}>
-        <Repository />
-        <Repository />
+        {repositories?.map(({ id, full_name, html_url, description }) => {
+          return (
```
Comment (remadyt) [inline src/widgets/Repositories/ui/Repositories.tsx]: obvious return not needed

**Review by remadyt — CHANGES_REQUESTED** [review]

**Pull request merged**